src/megatron/bridge/training/config.py

```diff
@@ -825,11 +825,13 @@ def validate(self) -> None:
         if self.peft is not None:
             assert self.checkpoint.pretrained_checkpoint is not None, "PEFT requires a pretrained checkpoint path"
 
-        data_seq_length = (
-            self.dataset.seq_length
-            if isinstance(self.dataset, FinetuningDatasetConfig)
-            else self.dataset.sequence_length
-        )
+        if isinstance(self.dataset, FinetuningDatasetConfig):
+            data_seq_length = self.dataset.sequence_length
+        elif self.dataset is not None:
+            data_seq_length = self.dataset.sequence_length
+        else:
+            data_seq_length = self.model.seq_length
```

Comment:
why do we need this case to work around the assertion in L835?

Reply:
this sets the correct fallback order. There are cases user are using custom dataset, and their dataset has not seq length.